d3a_api_client/redis_device.py

```diff
@@ -1,75 +1,20 @@
-
-    def _on_register(self, msg):
-        message = json.loads(msg["data"])
-        self.device_uuid = message['device_uuid']
-        self._check_buffer_message_matching_command_and_id(message)
-        logging.info(f"Client was registered to the device: {message}")
-        self.is_active = True
-        self.device_uuid = message['device_uuid']
-
-    def _on_unregister(self, msg):
-        message = json.loads(msg["data"])
-        self._check_buffer_message_matching_command_and_id(message)
-        logging.info(f"Client was unregistered from the device: {message}")
-        self.is_active = False
-
-    def _subscribe_to_response_channels(self, pubsub_thread=None):
-        if b'aggregator_response' in self.pubsub.patterns:
-            self._subscribed_aggregator_response_cb = self.pubsub.patterns[b'aggregator_response']
-        super()._subscribe_to_response_channels()
-
-    def _aggregator_response_callback(self, message):
-        if self._subscribed_aggregator_response_cb is not None:
-            self._subscribed_aggregator_response_cb(message)
-        data = json.loads(message['data'])
-        if data['transaction_id'] in self._transaction_id_buffer:
-            self._transaction_id_buffer.pop(self._transaction_id_buffer.index(data['transaction_id']))
-
-    def _check_transaction_id_cached_out(self, transaction_id):
-        return transaction_id in self._transaction_id_buffer
-
-    def select_aggregator(self, aggregator_uuid, is_blocking=True):
-        logging.info(f"Device: {self.device_id} is trying to select aggregator {aggregator_uuid}")
-
-        transaction_id = str(uuid.uuid4())
-        data = {"aggregator_uuid": aggregator_uuid,
-                "device_uuid": self.device_uuid,
-                "type": "SELECT",
-                "transaction_id": transaction_id}
-        self.redis_db.publish(f'aggregator', json.dumps(data))
-        self._transaction_id_buffer.append(transaction_id)
+        super().__init__(area_id, autoregister, redis_url, pubsub_thread)
 
-        if is_blocking:
-            try:
-                wait_until_timeout_blocking(
-                    lambda: self._check_transaction_id_cached_out(transaction_id)
-                )
-                logging.info(f"DEVICE: {self.device_uuid} has selected "
-                             f"AGGREGATOR: {aggregator_uuid}")
-                return transaction_id
-            except AssertionError:
-                raise RedisAPIException(f'API has timed out.')
+    def register(self, is_blocking=True):
+        super().register(is_blocking)
 
-    def unselect_aggregator(self, aggregator_uuid):
-        raise NotImplementedError("unselect_aggregator hasn't been implemented yet.")
+    def unregister(self, is_blocking=True):
+        super().unregister(is_blocking)
 
-    @property
-    def _channel_prefix(self):
-        return f"{self.device_id}"
+    def on_event_or_response(self, message):
```

Comment:
I think that `on_event_or_response` is not defined as an `@abstractmethod` on the interface, so it might be possible to remove it from here and from `RedisClientBase`. Or we might declar it as abstract on the interface, so that it will be clear that we want that method to be defined everywhere.

Reply:
Actually, it looks like this method doesn't have any implementation anywhere. Couldn't we just remove it?

Reply:
Indeed, you are right with your last comment. This in case the user likes to react in every event or repsonse.